executor/src/witgen/sequence_iterator.rs

```diff
@@ -201,15 +206,23 @@ impl Iterator for ProcessingSequenceIterator {
         match self {
             Self::Default(it) => it.next(),
             Self::Cached(it) => it.next(),
+            Self::Incomplete => unreachable!(),
         }
     }
 }
 
+enum CacheEntry {
+    /// The machine has been run successfully before and the sequence is cached.
+    Cached(Vec<SequenceStep>),
+    /// The machine has been run before, but did not succeed. There is no point in trying again.
+    Incomplete,
```

Comment:
This is a bit confusing: Why is there a cached incomplete and a non-cached incomplete?

Reply:
True, I renamed the first variant `Complete`. The idea is: If there is a cache entry, either it has been successful (and should use the cached sequence the next time) or it has not been successful, in this case it should abort without trying again.